data-prepper-plugins/s3-source/src/test/java/org/opensearch/dataprepper/plugins/source/configuration/S3SelectOptionsTest.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.source.configuration;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import java.lang.reflect.Field;
+
+import static org.hamcrest.CoreMatchers.sameInstance;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+class S3SelectOptionsTest {
+
+    @ParameterizedTest
+    @CsvSource({"csv","json","parquet"})
+    void s3SelectOptionsTest(final String dataSerializationFormat) throws NoSuchFieldException, IllegalAccessException {
+        S3SelectOptions s3SelectOptions = new S3SelectOptions();
+        final String queryStatement = "select * from s3Object";
+        final S3SelectSerializationFormatOption s3SelectSerializationFormatOption = S3SelectSerializationFormatOption.fromOptionValue(dataSerializationFormat);
+        reflectivelySetField(s3SelectOptions,"queryStatement",queryStatement);
+        reflectivelySetField(s3SelectOptions,"s3SelectSerializationFormatOption",s3SelectSerializationFormatOption);
+        assertThat(s3SelectOptions.getQueryStatement(),sameInstance(queryStatement));
+        assertThat(s3SelectOptions.getS3SelectSerializationFormatOption(),sameInstance(s3SelectSerializationFormatOption));
+    }
+    private void reflectivelySetField(final S3SelectOptions s3SelectOptions, final String fieldName, final Object value)
```

Comment:
Data Prepper has this as a utility in a test library. See: https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-test-common/src/main/java/org/opensearch/dataprepper/test/helper/ReflectivelySetField.java

Please use that.

Reply:
Thanks David for your review, Changes are incorporated and available for review.